src/services/ShillMessageDataService.ts

```diff
@@ -28,10 +28,24 @@ export class ShillMessageDataService {
       .all();
 
     records.forEach(record => {
-      items.push(this.shillMessage(record));
+      const content = record.get('content') as string;
+      const expiry = record.get('expiry') as string;
+      const currentDate = Date.now();
+      let expiredDate = -Infinity;
+
+      if (typeof expiry === 'string') {
+        expiredDate = Date.parse(expiry);
+      }
+
+      const hasContent = typeof content === 'string';
```

Comment:
I feel like having content and having valid content are two different things. You can have content and it be invalid (like it's a number, or an array, and not a string. Could we change this so that it is congruent with the meaning of the result of this comparison? Something like "isValidContent"?